stix_shifter_modules/guardium/stix_transmission/api_client.py

```diff
@@ -1,250 +1,73 @@
-                        str((self.authorization).get("access_token"))
-#
-# Test  -- uncomment to test token failure
-        #auth = "Bearer 575d161c-9a91-4c05-bf0a-1c2d415a8c40"
-        self.add_header_elements("authorization", auth)
-        return
-#
-# NOTE: connector architecture forces sync connector behave as async connector
-#       therefore, the state of the connector has to preserved in the search_id
-#       which is not generated by Guardium.  We generate and store necessary status_code
-#       such as self.credential, self.query (may be self.authorization)
-#       IF search_id originally generated is changed then the state is lost
-
-    def set_searchId(self, search_id):
-        self.search_id = search_id
-        return
-
-    def build_searchId(self):
-        #       It should be called only ONCE when transmit query is called
-        # Structure of the search id is
-        # '{"query": ' + json.dumps(self.query) + ', "credential" : ' + json.dumps(self.credential) + '}'
-        s_id = None
-
-        if(self.query is None or self.authorization is None or self.credential is None):
-            raise IOError(3001, 
-            "Could not generate search id because 'query' or 'authorization token' or 'credential info' is not available.")
-
-        else:
-            id_str = '{"query": ' + json.dumps(self.query) + ', "credential" : ' + json.dumps(self.credential) + '}'
-
-            id_byt = id_str.encode('utf-8')
-            s_id = base64.b64encode(id_byt).decode()
-            self.set_searchId(s_id)
-
-        return s_id
-
-    def decode_searchId(self):
-        # These value (self.credential, self.query) must be present.  self.authorization may not.
-        try:
-            id_dec64 = base64.b64decode(self.search_id)
-            jObj = json.loads(id_dec64.decode('utf-8'))
-        except:
-            raise IOError(
-                3001, "Could not decode search id content - " + self.search_id)
-
-        self.query = jObj.get("query", None)
-        self.credential = jObj.get("credential", None)
-        self.authorization = jObj.get("authorization", None)
-        return
-
-    def get_searches(self):
-        # CAN NOT be implemented for Guardium
-
-        endpoint = self.endpoint_start + "searches"
-        return self.client.call_api(endpoint, 'GET', timeout=self.search_timeout)
+        return ResponseWrapper(respObj)
+   
+    def delete_search(self, search_id):
+        # Optional since this may not be supported by the data source API
+        # Delete the search
+        return {"code": 200, "success": True}
 
     def create_search(self, query_expression):
-        # validate credential and create search_id.  No query submission -- Sync call
-        
         respObj = Response()
         respObj.code = "401"
         respObj.error_type = ""
         respObj.status_code = 401
-        if (self.get_accessToken()):
+        # print("query="+query_expression)
+        if (self.client.access_token):
```

Comment:
not sure why you have removed self.fetch_accessToken. And also this is wrong statement (self.client.access_token).